### PR TITLE
MXS-1891: dealloc named prepare should route to all

### PR DESCRIPTION
**Pull request by ybbct**
I am contributing the new code of the whole pull request, including one or
several files that are either new files or modified ones, under the BSD-new
license.

Because prepare statement route to all, so drop/dealloc prepare should route to all too;

### CODE REVIEW

**Review by jhnwkmn — COMMENTED** [review]
In principle this looks ok. However, the prepare type information should also be removed from the PS manager.

**Comment by jhnwkmn** [comment]
I think this looks good but I'll let Markus also take a look before merging.

File: server/modules/routing/readwritesplit/rwsplit_route_stmt.cc
Comment (markus456) [inline server/modules/routing/readwritesplit/rwsplit_route_stmt.cc]: This can be combined into the previous `if` statement, the two are mutually exclusive.
```
     if (qc_query_is_type(type, QUERY_TYPE_PREPARE_NAMED_STMT) ||
         qc_query_is_type(type, QUERY_TYPE_PREPARE_STMT))
     ...
   else  if (qc_query_is_type(type, QUERY_TYPE_DEALLOC_PREPARE))
```

**Review by markus456 — APPROVED** [review]
Looks good.

**Pull request merged**
